pkg/config/expand.go

```diff
@@ -829,6 +829,18 @@ func expandSimpleVariable(context varContext, trackModuleGraph bool) (string, er
 	case varRef.fromGroupID:
 		// intragroup reference can make direct reference to module output
 		expandedVariable = fmt.Sprintf("((module.%s.%s))", varRef.toModuleID, varRef.name)
+		if trackModuleGraph {
+			var found bool
+			for _, conn := range context.dc.moduleConnections[varRef.fromModuleID] {
+				if slices.Contains(conn.sharedVariables, varRef.name) {
```

Comment:
Let's check for `conn.kind != explicit`.

Reply:
Addressed on PR branch.